Source/Core/Common/TraversalServer.cpp

```diff
@@ -160,28 +161,20 @@ static sockaddr_in6 MakeSinAddr(const TraversalInetAddress& addr)
   return result;
 }
 
-static void GetRandomBytes(void* output, size_t size)
+static void GetRandomBytes(unsigned char* output, size_t size)
+
 {
-  static u8 bytes[8192];
-  static size_t bytesLeft = 0;
-  if (bytesLeft < size)
-  {
-    ssize_t rv = read(urandomFd, bytes, sizeof(bytes));
-    if (rv != sizeof(bytes))
-    {
-      perror("read from /dev/urandom");
-      exit(1);
-    }
-    bytesLeft = sizeof(bytes);
-  }
-  memcpy(output, bytes + (bytesLeft -= size), size);
+  static std::independent_bits_engine<std::default_random_engine, sizeof(unsigned char) * 8,
```

Comment:
Engine has not been seeded and so will be seeded with a default_seed, which would be the same on each run.

Unfortunately properly seeding the <random> engines is [not as straightforward as it seems](http://open-std.org/JTC1/SC22/WG21/docs/papers/2016/p0205r0.html).

In order to be correct one would have to do something akin to the following in the current version of the standard:

    // This provides the same interface as std::seed_seq does.
    struct random_device_seed_seq
    {
        template<typename It>
        void generate(It begin, It end)
        {
            for (; begin != end; ++begin)
            {
                *begin = m_device();
            }
        }
    private:
        std::random_device m_device;
    };

    static random_device_seed_seq s_seed_seq;
    // l-value reference to seed_seq required.
    static std::independent_bits_engine<std::mt19937_64, CHAR_BIT, unsigned char> s_engine{s_seed_seq};

Of course since we don't really care about randomness quality we could get away with one of the lazier but wrong methods mentioned in the paper, although the first and easiest method mentioned would only give 8 bits of entropy here.

Reply:
Thanks for sharing that proposal. I especially liked reading the reference on Melissa O'Neill's blog—sounds like she's really dug into this a lot.

My main takeaway is that `<random>` is sort of broken. Womp. 😢

I think the seed sequence you suggested is the best/easiest way to do this, so I've added it to the PR.